### PR TITLE
Simulated Action Tests

### LINKED ISSUE #1870: Add SimulatedTacticTestFixture
### Description of the task
We want to test tactics using the simulator

### Acceptance criteria

- [ ] make SimulatedTacticTestFixture and an example test

### Blocked By
https://github.com/UBC-Thunderbots/Software/pull/1896

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Added SimulatedActionTestFixture 
* Added SimulatedMoveActionTest
* Support visualization

### Testing Done
example move action test

### Resolved Issues
resolves https://github.com/UBC-Thunderbots/Software/issues/1870

### Length Justification
just over

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Overall LGTM.

Because the `SimulatedActionTestFixture` specifies what navigator/path planner it's using, it could diverge from what is used in STP/for gameplay. Just want to double check this is intended, and that the purpose of these tests will more be for action logic rather that doesn't rely on navigator specifics? I'd still imagine there may be cases where you want to make sure an `InterceptAction` actually is able to intercept the ball with the navigator that would be useful for gameplay. In theory if the robots couldn't intercept the ball after a navigator change the SimulatedPlayTests would catch it, but I'd imagine we'd have better coverage of each Action in these tests, so this would be the more likely place to rely on to catch those issues.

File: src/software/simulated_tests/validation_functions/robot_at_position_validation.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include "software/world/world.h"
+
+/**
+ * Checks if the robot represented by robot_id has arrived at the destination at
+ * any point in the test
+ * @param robot_id the ID of the robot in question, there must exist a robot
+ * for the given robot_id
+ * @param world_ptr the world pointer given by the simulator. Gets updated every tick
+ * @param destination The point to arrive at
+ * @param close_to_destination_threshold The threshold where the robot is considered close
```
Comment (garethellis0) [inline src/software/simulated_tests/validation_functions/robot_at_position_validation.h:13]: It's probably worth adding units to this.

**Review by garethellis0 — COMMENTED** [review]
Nothing much to add on top of Mathew here

**Comment by jonathanlew** [comment]
closed due to FSM project

**Pull request closed without merging**